OsmAnd/src/net/osmand/plus/views/layers/MapTileLayer.java

```diff
@@ -384,42 +431,61 @@ public void drawTileMap(Canvas canvas, RotatedTileBox tileBox, DrawSettings draw
 				}
 			}
 		}
-		
+
 		if (mainMap && !oneTileShown && !useInternet && warningToSwitchMapShown < 3) {
 			if (resourceManager.getRenderer().containsLatLonMapData(view.getLatitude(), view.getLongitude(), nzoom)) {
 				Toast.makeText(view.getContext(), R.string.switch_to_vector_map_to_see, Toast.LENGTH_LONG).show();
 				warningToSwitchMapShown++;
 			}
 		}
 	}
-	
+
 
 	@Override
 	public int getMaximumShownMapZoom() {
-		return map == null ? 20 : map.getMaximumZoomSupported() + OVERZOOM_IN;
+		return map == null ? SHOWN_ZOOM_MAX : map.getMaximumZoomSupported() + OVERZOOM_IN;
 	}
-	
+
 	@Override
 	public int getMinimumShownMapZoom() {
-		return map == null ? 1 : map.getMinimumZoomSupported();
+		return map == null ? SHOWN_ZOOM_MIN : map.getMinimumZoomSupported();
 	}
-		
+
 	@Override
 	public void destroyLayer() {
 		setMapTileAdapter(null);
+		isSetProvider = false;
 		if (resourceManager != null) {
 			resourceManager.removeMapTileLayerSize(this);
 		}
+
+		if (view != null) {
+			final MapRendererView mapRenderer = view.getMapRenderer();
+			if (mapRenderer != null) {
+				//mapRenderer.resetMapLayerProvider(getLayerId());
+			}
+		}
+
 	}
 
 	public boolean isVisible() {
 		return visible;
 	}
-	
+
 	public void setVisible(boolean visible) {
 		this.visible = visible;
+
+		if (!visible) {
+			if (view != null) {
```

Comment:
triple embedded if could be done within 1 if ?

Reply:
Will update.

Reply:
Logic have been changed/updated.